applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -16,6 +16,12 @@
 from KratosMultiphysics import auxiliary_solver_utilities
 import KratosMultiphysics.kratos_utilities as kratos_utils
 
+if kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication"):
+    import KratosMultiphysics.EigenSolversApplication as EiSA
+
+if kratos_utils.CheckIfApplicationsAvailable("ExternalSolversApplication"):
+    import KratosMultiphysics.ExternalSolversApplication as ExSA
```

Comment:
importing bcs you might need is is not allowed. The factory imports it if necessary
(by the way we had problems with this)